extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -238,7 +241,7 @@ export class AmpAutocomplete extends AMP.BaseElement {
     return this.mutateElement(() => {
       this.renderResults_();
       this.toggleResults_(true);
-    });
+    }).catch(e => this.renderFallbackUI_(e));
```

Comment:
I don't think this `catch` block is correctly placed since it is on the `mutateElement` call, which doesn't pass on an error thrown by `renderResults`. Where's a better place for it?

Reply:
I'm not sure I understand which of `renderResults` or `toggleResults` could fail here? I thought your previous statement line 218 would catch everything for now, until you start polling the publisher's server on keydown?

Also, not related to this PR but `mutateElement` should only contain the lines that actually write things in the DOM. Anything else, like the filtering logic, should happen synchronously. Mostly because it becomes very easy for a developer to inadvertently introduce layout thrashing, since it's too hard to know what context you're in.

Consider the following scenario: `inputHandler_` calls a `mutateElement` that calls `renderResults_` that calls `clearAllItems_`. Tomorrow, another developer makes an update and reads from the DOM in `clearAllItems`, not knowing it was called in a `mutateElement`, and they're causing performance issues :(